### PR TITLE
fix all warnings

### PR DESCRIPTION
**Pull request by harryfei**

### CODE REVIEW

File: rust_src/build.rs [hunk 1]
```diff
@@ -497,7 +497,7 @@ fn build_ignored_crates(path: &PathBuf) -> bool {
 fn generate_crate_exports(path: &PathBuf) -> Result<(), BuildError> {
     let modules = find_crate_modules(&path.join("src"))?;
 
-    fs::create_dir(path.join("out"));
+    let _ = fs::create_dir(path.join("out"));
```
Comment (brotzeit) [inline rust_src/build.rs:500]: Do you think it makes sense to return `BuildError` when `crate_dir` fails ?

Comment (harryfei) [reply rust_src/build.rs:500]: It makes sense.
I try it. If the directory exists, this creation return error too.  That breaks our build process.

**Comment by DavidDeSimone** [comment]
Does this new crate address https://github.com/emacs-ng/emacs-ng/issues/57, or is it's implantation of offset_of also unsound?

**Review by DavidDeSimone — APPROVED** [review]

**Comment by harryfei** [comment]
@DavidDeSimone yes

**Pull request merged**